### PR TITLE
KAA-411: Kaa's headers need to be installed into 'kaa' folder

### PR DESCRIPTION
**Pull request by ohamov**

### CODE REVIEW

File: client/client-multi/client-c/listfiles/CMakeGen.cmake [hunk 1]
```diff
@@ -15,8 +15,8 @@
 #
 
 set (KAA_SOURCE_FILES  ${KAA_SOURCE_FILES}
-                       src/kaa_event.c
-                       src/gen/kaa_profile_gen.c
-                       src/gen/kaa_logging_gen.c
-                       src/gen/kaa_configuration_gen.c
+                       src/kaa/kaa_event.c
```
Comment (ghost) [inline client/client-multi/client-c/listfiles/CMakeGen.cmake]: This file is auto-generated, it is created by routine, decribed in CSdkGenerator.java and CEventSourcesGenerator.java. So paths in these sources should be edited as well.

File: client/client-multi/client-c/CMakeLists.txt [hunk 2]
```diff
@@ -111,31 +111,31 @@ message("==================================")
...
-        src/kaa_channel_manager.c
-        src/kaa_common_schema.c
-        src/kaa_event.c
-        src/kaa_logging.c
-        src/kaa_profile.c
-        src/kaa_configuration_manager.c
-        src/kaa_status.c
-        src/kaa_user.c
-        src/kaa.c
+        src/kaa/avro_src/io.c
```
Comment (ghost) [inline client/client-multi/client-c/CMakeLists.txt]: I suggest to use the CMake variable to specify path to sources.

File: server/control/src/main/java/org/kaaproject/kaa/server/control/service/sdk/CSdkGenerator.java [hunk 3]
```diff
@@ -73,10 +77,10 @@
...
 
     private static final String KAA_CMAKEGEN         = "listfiles/CMakeGen.cmake";
-    private static final String KAA_DEFAULTS_HEADER  = "src/kaa_defaults.h";
-    private static final String PROFILE_HEADER       = "src/kaa_profile.h";
-    private static final String LOG_HEADER           = "src/kaa_logging.h";
-    private static final String CONFIGURATION_HEADER = "src/gen/kaa_configuration_definitions.h";
+    private static final String KAA_DEFAULTS_HEADER  = KAA_SRC_FOLDER + "/kaa_defaults.h";
+    private static final String PROFILE_HEADER       = KAA_SRC_FOLDER + "/kaa_profile.h";
+    private static final String LOG_HEADER           = KAA_SRC_FOLDER + "/kaa_logging.h";
+    private static final String CONFIGURATION_HEADER = KAA_SRC_FOLDER + "/gen/kaa_configuration_definitions.h";
```
Comment (ghost) [inline server/control/src/main/java/org/kaaproject/kaa/server/control/service/sdk/CSdkGenerator.java]: Please, use KAA_GEN_SOURCE_DIR instead.

**Comment by ghost** [comment]
LGTM

**Comment by izeigerman** [comment]
LGTM

**Pull request merged**